pallets/eth-bridge/src/util.rs

```diff
@@ -0,0 +1,177 @@
+    <T as pallet::Config>::TimeProvider::now().as_secs()
+}
+
+pub fn quorum_reached<T: Config>(entries: u32) -> bool {
+    let quorum = calculate_one_third_quorum(AVN::<T>::validators().len() as u32);
+    entries >= quorum
+}
+
+pub fn add_new_tx_request<T: pallet::Config>(
+    tx_id: u32,
+    tx_data: TransactionData<T>,
+) -> Result<(), Error<T>> {
+    let corroborations =
+        CorroborationData { tx_succeeded: BoundedVec::default(), tx_failed: BoundedVec::default() };
+
+    Transactions::<T>::insert(tx_id, tx_data);
+    Corroborations::<T>::insert(tx_id, corroborations);
+    UnresolvedTxs::<T>::try_mutate(|txs| {
+        txs.try_push(tx_id).map_err(|_| Error::<T>::UnresolvedTxLimitReached)
+    })
+}
+
+pub fn assign_sender<T: Config>() -> Result<T::AccountId, Error<T>> {
+    let current_block_number = <frame_system::Pallet<T>>::block_number();
+
+    match AVN::<T>::calculate_primary_validator(current_block_number) {
+        Ok(primary_validator) => {
+            let sender = primary_validator;
+            Ok(sender)
+        },
+        Err(_) => Err(Error::<T>::ErrorAssigningSender),
+    }
+}
+
+pub fn update_confirmations<T: Config>(
+    tx_id: u32,
+    confirmation: &ecdsa::Signature,
+    author: &Author<T>,
+) -> Result<(), Error<T>> {
+    let tx_data = Transactions::<T>::get(tx_id).ok_or(Error::<T>::TxIdNotFound)?;
+
+    if !UnresolvedTxs::<T>::get().contains(&tx_id) ||
+        author.account_id == tx_data.sender ||
+        quorum_reached::<T>(tx_data.confirmations.len() as u32 + 1)
+    {
+        return Ok(())
+    }
+
+    eth::verify_signature::<T>(tx_data.msg_hash, &author, &confirmation)?;
+    ensure!(!tx_data.confirmations.contains(confirmation), Error::<T>::DuplicateConfirmation);
+
+    let mut updated_tx_data = tx_data;
+    updated_tx_data
+        .confirmations
+        .try_push(confirmation.clone())
+        .map_err(|_| Error::<T>::ExceedsConfirmationLimit)?;
+
+    Transactions::<T>::insert(tx_id, updated_tx_data);
+
+    Ok(())
+}
+
+pub fn set_eth_tx_hash<T: Config>(
+    tx_id: u32,
+    eth_tx_hash: H256,
+    author: &Author<T>,
+) -> Result<(), Error<T>> {
+    let mut tx_data = Transactions::<T>::get(tx_id).ok_or(Error::<T>::TxIdNotFound)?;
+    ensure!(tx_data.eth_tx_hash == H256::zero(), Error::<T>::EthTxHashAlreadySet);
+    ensure!(tx_data.sender == author.account_id, Error::<T>::EthTxHashMustBeSetBySender);
+    tx_data.eth_tx_hash = eth_tx_hash;
+    Transactions::<T>::insert(tx_id, tx_data);
+    Ok(())
+}
+
+pub fn requires_corroboration<T: Config>(tx_id: u32, author: &Author<T>) -> Result<bool, Error<T>> {
+    let corroboration =
+        Corroborations::<T>::get(tx_id).ok_or_else(|| Error::<T>::CorroborationNotFound)?;
+    let not_in_succeeded = !corroboration.tx_succeeded.contains(&author.account_id);
+    let not_in_falied = !corroboration.tx_failed.contains(&author.account_id);
+    Ok(not_in_succeeded && not_in_falied)
+}
+
+pub fn update_corroborations<T: Config>(
```

Comment:
I'd change the function name to describe what it does. Which seems to be: `add_author_to_corroborations`

Reply:
It was named this way partly to differentiate it from the "add_" methods which relate to the extrinsic calls, and partly because it doesn't necessarily add the author to the corroborations, which depend on the state at the time.